db/src/tx.rs

```diff
@@ -412,6 +421,26 @@ impl<'conn, 'a, W> Tx<'conn, 'a, W> where W: TransactWatcher {
                                 Either::Right(LookupRefOrTempId::LookupRef(in_process.intern_lookup_ref(lookup_ref)?))
                             },
 
+                            entmod::AtomOrLookupRefOrVectorOrMapNotation::TxFunction(ref tx_function) => {
+                                let typed_value = match tx_function.op.0.as_str() {
+                                    "current-tx" => TypedValue::Ref(self.tx_id),
+                                    unknown @ _ => bail!(ErrorKind::NotYetImplemented(format!("Unknown transaction function {}", unknown))),
+                                };
+
+                                // Here we do schema-aware typechecking: we assert that the computed
+                                // value is in the attribute's value set.  If and when we have
+                                // transaction functions that produce numeric values, we'll have to
+                                // be more careful here, because a function that produces an integer
+                                // value can be used where a double is expected.  See also
+                                // `SchemaTypeChecking.to_typed_value(...)`.
```

Comment:
The algebrizer function stuff has some coercions that might be useful to extract.

Reply:
Thanks for the quick response.  I do expect there's some shared typechecking stuff between query and parse; they're getting closer, not further apart :)